lib/matplotlib/backends/backend_pdf.py

```diff
@@ -1700,8 +1700,19 @@ def _writeImg(self, data, id, smask=None):
                 # Convert to indexed color if there are 256 colors or fewer
                 # This can significantly reduce the file size
                 num_colors = len(img_colors)
-                img = img.convert(mode='P', dither=Image.NONE,
-                                  palette=Image.ADAPTIVE, colors=num_colors)
+                # These constants were converted to IntEnums and deprecated in
+                # Pillow 9.2
+                dither = (
+                    Image.Dither.NONE if hasattr(Image, 'Dither')
+                    else Image.NONE
+                )
+                pmode = (
+                    Image.Palette.ADAPTIVE if hasattr(Image, 'Palette')
+                    else Image.ADAPTIVE
+                )
```

Comment:
A shorter (but perhaps more confusing?) way to write it would be:
```suggestion
                dither = getattr(Image, 'Dither', Image).NONE
                pmode = getattr(Image, 'Palette', Image).ADAPTIVE
```

Reply:
I'm 50/50.  If anyone else likes this better feel free to commit it.

Reply:
fwiw I think the shorter version is fine (not so much because it's more *or* less legible, but rather because anyways I tend to read these as "bah, that's just whatever is needed to handle accessing NONE across versions, I don't really care about pillow's API changes...").